---
title: "data_curation"
output: html_document
---

The following obtains the raw sequences from the Sequence Read Archive (SRA) and renames the fastq files. The file sra_to_samples.txt has the accession numbers and identifiers of the paired reads from each sample. The identifiers are in the same format obtained from the Illumina Miseq sequencer:
\<SampleName>_S\<SampleNumber>_L001_R\<ReadNumber>_001.
The mapping_file.txt contains a link from the sample ID to the metadata.

Demultiplexing was done automatically using the onboard PC of the Miseq instrument and sequences represent only biological data (i.e. do not contain adapters or primers sequences). 

After retrieving and renaming the data, the raw files will be located in the fastq_files directory within the rendered_Rmds directory.

```{r, engine='bash', results='hide'}
#download raw sequences from SRA
wget --quiet -r --no-parent ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP098/SRP098921/
mv ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP098/SRP098921/*/*.sra ./
rm -rf ftp-trace.ncbi.nih.gov/

#dump fastq files sra files
for sra in *.sra
do
  sratoolkit.2.8.1-2-centos_linux64/bin/fastq-dump --split-3 "$sra" &
done
wait

rm -f *.sra

#rename files using sample identifiers used throughout analysis
while read sra new
do
    for srafile in $sra*
    do
        newfile="$new""$(echo "$srafile" | cut -c11-)"
        mv "$srafile" "$newfile" 
    done
done < sra_to_samples.txt

mkdir rendered_Rmds
mkdir rendered_Rmds/fastq_files
mv *.fastq rendered_Rmds/fastq_files
```